tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_friendship.c

```diff
@@ -616,6 +662,65 @@ static void test_lpn_overflow(void)
 		     msg.seq, msg.ctx.addr);
 	}
 
+	LOG_INF("Testing overflow with unsegmented message preempting segmented one...");
+
+	ASSERT_OK(bt_mesh_lpn_poll(), "Poll failed");
+
+	/* Last seq from the previous step. */
+	exp_seq = CONFIG_BT_MESH_FRIEND_QUEUE_SIZE + 1;
+
+	exp_seq += 2; /* Skipping the first message in Friend Queue. */
+	ASSERT_OK(bt_mesh_test_recv_msg(&msg, K_SECONDS(2)),
+		  "Receive first unseg msg failed");
+	ASSERT_EQUAL(5, msg.len);
+	ASSERT_EQUAL(cfg->addr, msg.ctx.recv_dst);
+	ASSERT_EQUAL(exp_seq, msg.seq);
+
+	exp_seq++;
+	ASSERT_OK(bt_mesh_test_recv_msg(&msg, K_SECONDS(2)),
+		  "Receive the second unseg msg failed");
+	ASSERT_EQUAL(5, msg.len);
+	ASSERT_EQUAL(cfg->addr, msg.ctx.recv_dst);
+	ASSERT_EQUAL(exp_seq, msg.seq);
+
+	/* Not expecting any more messages from friend */
+	err = bt_mesh_test_recv_msg(&msg, K_SECONDS(10));
+	if (!err) {
+		FAIL("Unexpected additional message 0x%02x from 0x%04x",
+		     msg.seq, msg.ctx.addr);
+	}
+
+	LOG_INF("Testing overflow with segmented message preempting another segmented...");
+
+	ASSERT_OK(bt_mesh_lpn_poll(), "Poll failed");
+
+	exp_seq += 2; /* Skipping the first message in Friend Queue. */
+	ASSERT_OK(bt_mesh_test_recv_msg(&msg, K_SECONDS(2)),
+		  "Receive the first unseg msg failed");
+	ASSERT_EQUAL(5, msg.len);
+	ASSERT_EQUAL(cfg->addr, msg.ctx.recv_dst);
+	ASSERT_EQUAL(exp_seq, msg.seq);
+
+	exp_seq++;
+	ASSERT_OK(bt_mesh_test_recv_msg(&msg, K_SECONDS(20)),
+		  "Receive the seg msg failed");
+	ASSERT_EQUAL(BT_MESH_SDU_UNSEG_MAX * (CONFIG_BT_MESH_FRIEND_QUEUE_SIZE - 2),
+		     msg.len);
+	ASSERT_EQUAL(cfg->addr, msg.ctx.recv_dst);
+	ASSERT_EQUAL(exp_seq, msg.seq);
+
+	ASSERT_OK(bt_mesh_test_recv_msg(&msg, K_SECONDS(2)),
+		  "Receive the second unseg msg failed");
```

Comment:
FYI This macro doesn't print text message. This is bug of ASSERT_OK.

Reply:
I added analogue of `ASSERT_OK` macro that will print message: https://github.com/zephyrproject-rtos/zephyr/pull/53143/commits/49fee4f6db15373e70ba76bc59eebfbdb08ebc7f